eng/scripts/Generate-Patch.ps1

```diff
@@ -0,0 +1,240 @@
+# 5. BranchName             - The name of the remote branch where the patch changes will be pushed. This is not a required parameter.
+#                             In case the argument is not provided the branch name is release/{ArtifactName}_{ReleaseVersion}.
+#                             The script pushes the branch to remote URL https://github.com/Azure/azure-sdk-for-java.git
+#
+# Example:  .\eng\scripts\Generate-Patch.ps1 -ArtifactName azure-mixedreality-remoterendering -ServiceDirectory remoterendering -ReleaseVersion 1.0.0 -PatchVersion 1.0.1
+# This creates a remote branch "release/azure-mixedreality-remoterendering" with all the necessary changes.
+
+param(
+  [Parameter(Mandatory=$true)][string]$ArtifactName,
+  [Parameter(Mandatory=$true)][string]$ServiceDirectoryName,
+  [Parameter(Mandatory=$true)][string]$ReleaseVersion,
+  [Parameter(Mandatory=$false)][string]$PatchVersion,
+  [Parameter(Mandatory=$false)][string]$BranchName
+)
+
+function TestPathThrow($Path) {
+  if(!(Test-Path $Path)) {
+   LogError "$Path not found. Exiting ..."
+   exit
+  }
+}
+
+Write-Information "PS Script Root is: $PSScriptRoot"
+Write-Information "ArtifactName is: $ArtifactName"
+Write-Information "ReleaseVersion is: $ReleaseVersion"
+Write-Information "ServiceDirectoryName is: $ServiceDirectoryName"
+
+$MainRemoteUrl = 'https://github.com/Azure/azure-sdk-for-java.git'
+$RepoRoot = Resolve-Path "${PSScriptRoot}..\..\.."
+$EngDir = Join-Path $RepoRoot "eng"
+$EngCommonScriptsDir = Join-Path $EngDir "common" "scripts"
+$SdkDirPath = Join-Path $RepoRoot "sdk"
+$ServiceDirPath = Join-Path $SdkDirPath $ServiceDirectoryName
+$ArtifactDirPath = Join-Path $ServiceDirPath $ArtifactName
+$ArtifactPomFile = Join-Path $ArtifactDirPath "pom.xml"
+$ReleaseTag = -join($ArtifactName, "_", $ReleaseVersion)
+$TestResourcesFilePath = Join-Path $ServiceDirPath "test-resources.json"
+$GroupId = "com.azure"
+$EngVersioningDir = Join-Path $EngDir "versioning"
+$SetVersionFilePath = Join-Path $EngVersioningDir "set_versions.py"
+$UpdateVersionFilePath = Join-Path $EngVersioningDir "update_versions.py"
+$CodeQualityReports = Join-Path $EngDir "code-quality-reports" "src" "main" "resources"
+$CheckStyleSuppressionFilePath = Join-Path $CodeQualityReports "checkstyle" "checkstyle-suppressions.xml"
+$SpotBugsFilePath = Join-Path $CodeQualityReports "spotbugs" "spotbugs-exclude.xml"
+
+TestPathThrow -Path $ArtifactDirPath
+TestPathThrow -Path $EngDir
+TestPathThrow -Path $EngCommonScriptsDir
+TestPathThrow -Path $ArtifactDirPath
+TestPathThrow -Path $ArtifactPomFile
+TestPathThrow -Path $SdkDirPath
+TestPathThrow -Path $SetVersionFilePath
+TestPathThrow -Path $UpdateVersionFilePath
+
+. (Join-Path $EngCommonScriptsDir common.ps1)
+
+function GetPatchVersion($ReleaseVersion) {
+  $REGEX_VERSION = '([0-9]+).([0-9]+).([0-9]+)'
```

Comment:
Are beta versions interesting to handle?

Reply:
In theory, we can do a patch release for beta libraries too and so this can be generic but I have scoped it  for now as we currently only need this for GA libraries.